### PR TITLE
Use perform_action instead of dock() for cleaning tasks

### PR DESCRIPTION
**Pull request by xiyuoh**
This PR follows the changes made for rmf-web in https://github.com/open-rmf/rmf-web/pull/643. It also updates how performable actions are added to the rmf demo fleet adapter.

This PR:
- Adds a `check_perform_action()` to RobotCommandHandle to process performable actions such as cleaning tasks
- Removes `dock_name` from cleaning zones in hotel and airport building.yaml
- Adds an `action` field to the relevant worlds' config.yaml to specify `clean` as a performable action, and updates fleet_adapter accordingly to add them to the fleet if the field is populated
- Updates `dispatch_clean` to publish an ApiRequest with a compose task json

### CODE REVIEW

**Comment by Yadunund** [comment]
Suggest to refactor as below so that you can also update the schedule itinerary of the robot while cleaning. This is a feature that is currently supported in the fleet adapters we have implemented for other cleaning robots which relies on docking.
1. Add YAML nodes containing waypoints for processes to Fleet [config](https://github.com/open-rmf/rmf_demos/blob/7c6cef418393db843444d661869a10a096ca9448/rmf_demos/config/hotel/cleanerBotA_config.yaml#L40)
```
process_waypoints:
  zone_1: [[x1,y1], ... [xn, yn]
  zone_2: [[x1,y1], ...[xn, yn]
```

2. Add an [API](https://github.com/open-rmf/rmf_demos/blob/7c6cef418393db843444d661869a10a096ca9448/rmf_demos_fleet_adapter/rmf_demos_fleet_adapter/RobotClientAPI.py#L94) to the fleet manager to retrieve the list of waypoints for a given provess str. (the fleet_manager will return the list populated as described in Step 1)

3. Update [start_process()](https://github.com/open-rmf/rmf_demos/blob/7c6cef418393db843444d661869a10a096ca9448/rmf_demos_fleet_adapter/rmf_demos_fleet_adapter/fleet_manager.py#L289) in fleet_manager to publish path request with waypoint locations from the `process_waypoints` yaml elements in the config.

4. Update the `action_executor` such that when it parses an action, it
  a. Calls the API from 2) to get the waypoints for the process.
  b. Calls the `start_process()` api to trigger the fleet manager to send the path request to slotcar
  c. Use the unstable participant API to update RMF traffic schedule with the cleaning path of the robot. ie take the result of a) and call the `participant.set_itinerary()` API. This step is important for real world deployments where we want other robots in the facility to be aware of the cleaning action. This has already been implemented within perform_action for the Ecobot fleet adapter which you can reference [here](https://github.com/open-rmf/fleet_adapter_ecobot/blob/aae8d30b75c368e46784b4387584b36a503d1ac6/fleet_adapter_ecobot/EcobotCommandHandle.py#L543)
  
  This way, there is no explicit mention of `clean_` in  any of the APIs here. We keep things process and vendor neutral. It also does not request a cleaning server.

**Comment by luca-della-vedova** [comment]
I think there might be an issue with the update of the RMF schedule, i.e. below:

https://user-images.githubusercontent.com/9111558/212003134-85328bc8-82d7-4428-9c31-dc01bfc6ae21.mp4

I think it is due to the fact that the schedule submitted in every iteration is always the same (initial schedule fetched through the `retrieve_process_waypoints` API), we should probably keep track of the process waypoints, the position of the robot and update the schedule accordingly over time, to make sure traffic deconfliction works as intended

File: rmf_demos_fleet_adapter/rmf_demos_fleet_adapter/RobotCommandHandle.py [hunk 1]
```diff
@@ -543,6 +608,7 @@ def update_state(self):
                 if not self.started_action:
                     self.started_action = True
                     self.api.toggle_action(self.name, self.started_action)
+                    self.check_perform_action()
```
[the remark this answers is not in the document]
Comment (xiyuoh) [reply rmf_demos_fleet_adapter/rmf_demos_fleet_adapter/RobotCommandHandle.py]: agreed, updated in [96fff51](https://github.com/open-rmf/rmf_demos/pull/162/commits/96fff51c82d311c36375002ac89b94360aed1eb5)

File: rmf_demos_fleet_adapter/rmf_demos_fleet_adapter/RobotCommandHandle.py [hunk 2]
```diff
@@ -476,6 +484,63 @@ def _dock():
...
+                if self._quit_action_event.wait(0.1):
+                    self.node.get_logger().info("Aborting action")
+                    return
+
+            # End the robot action once it is completed
+            self.complete_robot_action()
+
+        self._action_thread = threading.Thread(target=_perform_action)
+        self._action_thread.start()
+        return
```
[the remark this answers is not in the document]
Comment (xiyuoh) [reply rmf_demos_fleet_adapter/rmf_demos_fleet_adapter/RobotCommandHandle.py]: [96fff51](https://github.com/open-rmf/rmf_demos/pull/162/commits/96fff51c82d311c36375002ac89b94360aed1eb5)

File: rmf_demos/config/airport_terminal/cleanerBotE_config.yaml [hunk 3]
```diff
@@ -37,6 +37,40 @@ rmf_fleet:
     delivery: False
     clean: True
     finishing_request: "park" # [park, charge, nothing]
+    action: ["clean"]
```
[the remark this answers is not in the document]
Comment (xiyuoh) [reply rmf_demos/config/airport_terminal/cleanerBotE_config.yaml]: Changed to `actions` in [96fff51](https://github.com/open-rmf/rmf_demos/pull/162/commits/96fff51c82d311c36375002ac89b94360aed1eb5). `clean: True` is not used anymore, removed it from all the configs and fleet_adapter.py in [fdb2d28](https://github.com/open-rmf/rmf_demos/pull/162/commits/fdb2d282dabe362430a91239494be581c98aa0ad)

**Comment by mxgrey** [comment]
The relevant parts of this PR have been merged into https://github.com/open-rmf/rmf_demos/pull/188 and the cleaning activity should be working nicely when combined with https://github.com/open-rmf/rmf_ros2/pull/286

I'll close this PR now since it's no longer needed.

**Pull request closed without merging**